core/test/scenarios/api-test-pptr.js

```diff
@@ -144,6 +144,49 @@ describe('Fraggle Rock API', function() {
 
       expect(erroredAudits).toHaveLength(0);
     });
+
+    // eslint-disable-next-line max-len
+    it('should know target type of network requests from frames created before timespan', async () => {
+      state.server.baseDir = `${LH_ROOT}/cli/test/fixtures`;
+      const {page, serverBaseUrl} = state;
+
+      page.goto(`${serverBaseUrl}/oopif-scripts-timespan.html`);
+      const run = await api.startTimespan(state.page);
```

Comment:
Doing a `page.goto` and `api.startTimespan` concurrently is not a recipe for consistent test results, and probably isn't a good practice for timespan mode in general.

Instead of loading the workers on a delay, wdyt about loading them after clicking a button and then add an action to click that button during the timespan?

Reply:
good idea